mimic/rest/maas_api.py

```diff
@@ -30,6 +30,16 @@
 Request.defaultContentType = 'application/json'
 
 
+def _find_index_of_id(xs, val):
```

Comment:
There is a built-in method on `list`, "index", which I think you want to use instead.  It raises an exception (`IndexError`) rather than returning a sentinel (`-1`), but since you're explicitly testing against `-1` anyway (without documenting it as a constant, I might add ;-)) catching the exception might be better.